src/lib.rs

```diff
@@ -99,16 +99,62 @@ impl PartialEq for NamespacedDataResponse {
     }
 }
 
+#[derive(Debug)]
+struct CelestiaNodeEndpoints {
+    namespaced_data: Url,
+    namespaced_shares: Url,
+    submit_pfd: Url,
+}
+
+impl CelestiaNodeEndpoints {
+    fn try_from_url(url: &Url) -> eyre::Result<Self> {
+        let namespaced_data = url
+            .join(NAMESPACED_DATA_ENDPOINT)
+            .wrap_err("failed creating URL for namespaced data endpoind")?;
```

Comment:
```suggestion
            .wrap_err("failed creating URL for namespaced data endpoint")?;
```

same for others below

Reply:
what would a potental error case be here actually? can `Url` be malformed?